sphinx/writers/latex.py

```diff
@@ -383,28 +415,34 @@ def cell(self, row=None, col=None):
 
 
 class TableCell(object):
+    """A cell data of tables."""
+
     def __init__(self, table, row, col):
         if table.cells[(row, col)] == 0:
             raise IndexError
 
         self.table = table
         self.cell_id = table.cells[(row, col)]
-        for n in range(row + 1):
-            if table.cells[(row - n, col)] == self.cell_id:
-                self.row = row - n
-        for n in range(col + 1):
-            if table.cells[(row, col - n)] == self.cell_id:
-                self.col = col - n
+        self.row = row
+        self.col = col
+
+        # adjust position for multirow/multicol cell
+        while table.cells[(self.row - 1, self.col)] == self.cell_id:
+            self.row -= 1
+        while table.cells[(self.row, self.col - 1)] == self.cell_id:
+            self.col -= 1
```

Comment:
it's only because in TeX macro takes at least 1000 times more than in any other language so I am always on look out for redundancy ... anyway my proposed alternative doubled Boolean checks so you have convinced me !